index.html

```diff
@@ -507,7 +512,66 @@ <h3>StatusList2021Credential</h3>
 <a href="#bitstring-encoding"></a>.
               </td>
             </tr>
-
+            <tr>
+              <td>
+                credentialStatus.ttl
+              </td>
+              <td>
+                The <code>ttl</code> indicates the "time to live" in milliseconds.
+                This property MAY be present, but if not present, implementers MUST 
+                use a value of <code>300000</code> for this property.  A verifier
+                MUST NOT use a cached <code>StatusList2021Credential</code> that was
+                cached more for more than the <code>ttl</code> duration prior to the
+                start of verification operation on a <a>verifiable credential</a>.
+                Implementations that publish the status list SHOULD align
+                any protocol-specific caching information, such as the 
+                HTTP `Cache-Control` header, with the value in this field.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                credentialStatus.size
+              </td>
+              <td>
+                The <code>size</code> indicates the size of the status entry in bits.
+                If size is omitted then the size of each status entry is assumed to be
+                1 bit (and therefore capable of capturing a boolean status).
+                <code>size</code> MAY be provided, but if <code>size</code> is not present
+                as a property of the <code>credentialStatus</code> then <code>size</code>
+                MUST be processed as 1.  <code>size</code> MUST be greater than zero.
+                if <code>size</code> is provided, and is greater than 1, then the property
+                <code>credentialStatus.message</code> MUST be present, and the number of
+                status messages must equal the number of possible values.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                credentialStatus.message
+              </td>
+              <td>
+                The <code>message</code> property MUST be an array, and if present
```

Comment:
This seems more like an enumeration table or something like that, I'm confused by "message" being used as the name. Shouldn't this be defined in the status list credential instead of here? The same goes for `size` ... right? It seems like that should be defined in one place in the status list credential. Why have this in each status entry?

I haven't implemented this -- so these are just questions that come to mind.

Reply:
Thinking about this table more -- why is it useful to have it in the VC? I see that there's some text below about adding other values -- is this to, for example, provide human readable descriptions?

Reply:
Couple of reasons - one human readable - two the intended status messages and values at a point in time can be signed and verified, three it simplifies message handling when verifying and processing the status check

Reply:
I would think that there could be a tendency for processing software to break if it depends on specific values for specific issuers -- and the human readable forms change. Of course, this will hopefully just pressure issuers not to make those changes, but could it be better for verification software to use the status numerical values and not the human readable ones -- keeping their own table internally for that?

I'd also think that verifiers will need to read some kind of documentation to understand and customize the processing of each individual issuer. Perhaps we want to externalize the enumeration so that values could be shared? It seems like it might have scalability issues otherwise.

It also seems that it might create challenges for writing middleware software that performs verification using generalized processing, as it doesn't know what to do with the status results. These will just need to be returned now -- which changes some existing abstractions, I think.

I imagine that checking status for these types of things will always be either:

1. It 'passed' because the status value could be retrieved; the value was 'foo'.
2. It 'failed' because the status couldn't be retrieved.

And we'll need some API space in various places to ensure that status values (numerical or strings, whatever is decided) can be returned. In short, checking a status list will no longer be about "is it on the list or not" and failing a check if it's been flagged (on the list) and passing if not -- which presently fits into existing generalized verification architecture and middleware. In short, I think this change may impact the semantics of what it means for a "status check to succeed" as [mentioned in the definition of "verification" in the VCDM today](https://www.w3.org/TR/vc-data-model/#dfn-verify). At the very least, it moves the abstraction bar a little bit. That isn't a statement of "good" or "bad", just of another difference in complexity and / or where the complexity lives or how it is expressed.

Reply:
part of the reason that the message array permits additional properties in addition to a status message along side the code itself, is that this permits system specific processing rules or hints to be embedded with status as well as things like messaging some examples follow...

a minimal example with the way things are written now:
```
"message": [
  {"status":"0x00", "value":"valid"},
  {"status":"0x01", "value":"invalid"},
  {"status":"0x02", "value":"hold"},
  ...
]
```

but this could obviously be extended with hints around whether to verify a credential (if that were appropriate for the use case - not all statuses are related to whether or not to verify a credential):
```
"message": [
  {"status":"0x00", "value":"valid", "verify": true},
  {"status":"0x01", "value":"invalid", "verify": false},
  {"status":"0x02", "value":"hold", "verify": true},
  ...
]
```

as the above example shows, we may want to provide the ability for a `verify` property in a status message that is optional, so that folks handle that in a standardized way, but I would prefer not to go down that path until we have a baseline merged in that permits general status to be tracked with multiple values

you could also go in and do some nifty stuff further and link out to rules defined in a rules engine etc with additional properties, but those would be very system and implementation dependent.

as I noted in another comment, the attachment of a code and a stable or regulatory defined message along side it is very common in regulatory environments.

If we want to limit this work item to verification rules, then we should retitle the overall work item to "verification list" and take this pull request and put it in a different work item to reflect actual status tracking in addition to verification capabilities.

Reply:
> If we want to limit this work item to verification rules, then we should retitle the overall work item to "verification list" and take this pull request and put it in a different work item to reflect actual status tracking in addition to verification capabilities.

I wouldn't be opposed to this. It's up to the WG to define how we publish work items. The concern here is on the verification complexity that this feature might add. It is generalizable (via inclusion of `valid`), but it doesn't seem like that's a design goal?